vhost_user_backend/src/lib.rs

```diff
@@ -113,6 +113,40 @@ pub trait VhostUserBackend: Send + Sync + 'static {
     /// A default implementation is provided as we cannot expect all backends
     /// to implement this function.
     fn set_slave_req_fd(&mut self, _vu_req: SlaveFsCacheReq) {}
+
+    /// Get epoll number.
+    /// While multiple queue are defined, it requires the backend to tell how many
+    /// vring handler thread it would like to run. The epoll number is not always
+    /// equal to the queue number, for net, it will be the count of queue pairs.
+    fn epoll_number(&self) -> usize;
+
+    /// Get vring subset.
+    /// While run multiple vring handler thread to service the multiple queues,
+    /// it requires the backend to how many queues each vring handler will handle.
+    fn vring_subset(
+        &self,
+        index: usize,
+        vrings: Vec<Arc<RwLock<Vring>>>,
+    ) -> Vec<Arc<RwLock<Vring>>>;
+
+    /// Get worker thread.
+    /// Get the worker threads from backends if they create internally to handle
+    /// vring events.
+    fn vring_worker_thread(
+        &self,
+    ) -> Option<Vec<thread::JoinHandle<result::Result<(), io::Error>>>> {
```

Comment:
Maybe this can be removed if we don't think there's a use case where the backend implementation would create its own worker thread.